src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -271,6 +271,36 @@ public void testAsyncAndSynchronousQueries() throws SQLException {
     con.close();
   }
 
+  /**
+   * Tests that error message and error code are reset after an error
+   *
+   * @throws SQLException
+   * @throws InterruptedException
+   */
+  @Test
+  public void testQueryStatusErrorMessageAndErrorCode() throws SQLException, InterruptedException {
+    // open connection and run asynchronous query
+    Connection con = getConnection();
+    Statement statement = con.createStatement();
+    statement.execute("create or replace table testTable(colA string, colB boolean)");
+    statement.execute("insert into testTable values ('test', true)");
+    ResultSet rs1 =
+        statement.unwrap(SnowflakeStatement.class).executeAsyncQuery("select * from testTable");
+    QueryStatus status = rs1.unwrap(SnowflakeResultSet.class).getStatus();
+    // Set the error message and error code so we can confirm they are reset when getStatus() is
+    // called.
+    status.setErrorMessage(QueryStatus.FAILED_WITH_ERROR.toString());
+    status.setErrorCode(2003);
+    status = rs1.unwrap(SnowflakeResultSet.class).getStatus();
+    // Assert status of query is a success
+    assertEquals(QueryStatus.SUCCESS, status);
```

Comment:
Looks good, except that the jenkins test server is a bit slow sometimes so this is coming back as RUNNING instead of SUCCESS. Putting in a brief thread sleep should fix this and prevent flaky tests. Something like a 3 second sleep timer.